docs/api.md

```diff
@@ -0,0 +1,531 @@
+## ArduinoCloud Class (Base)
+
+### `ArduinoCloud (Class)`
+
+#### Description
+
+`ArduinoIoTCloud` is a class for interfacing with the Arduino IoT Cloud. It provides an interface for managing IoT Cloud connectivity, updating data, and handling events.
+
+Depending on what type of connection is used (TCP/LPWAN), either the `ArduinoIoTCloudTCP` or `ArduinoIoTCloudLPWAN` base classes are initialized. This specification is done in the Arduino IoT Cloud's "Thing" interface, and is reflected in the automatically generated `thingProperties.h` file. 
+
+The above classes are inherited from the `ArduinoIoTCloudClass` base class, with methods used depending on what connection is used.
+
+#### Syntax
+
+```
+ArduinoCloud.method()
+```
+
+### `push()`
+
+#### Description
+
+Pushes data to the IoT Cloud.
+
+#### Syntax
+
+```
+ArduinoCloud.push()
+```
+
+#### Parameters
+
+None.
+
+#### Returns
+
+Nothing.
+
+### `setTimestamp()`
+
+Sets a timestamp for a cloud variable.
+
+#### Syntax
+
+```
+ArduinoCloud.setTimestamp(cloudVariable, timestamp)
+```
+
+#### Parameters
+
+- `String` - property name
+- `unsigned long const` - timestamp
+
+### `setThingId()`
```

Comment:
This funciton is only needed by LoRa devices, since for all TCP devices the ThingId is automaticallly configured by the Cloud via ThingId discovery protocol.